src/local-cache.ts

```diff
@@ -0,0 +1,84 @@
+import { del as idbDel, get as idbGet, set as idbSet } from 'idb-keyval';
+import { addSeconds } from './add-seconds';
+import type { Seconds } from './models';
+
+export interface LocalCacheInterface {
+  /**
+   * Set a cache with a ttl, in seconds
+   *
+   * @param {{ key: string; value: any; ttl?: Seconds }} options
+   * @returns {Promise<void>}
+   * @memberof LocalCacheInterface
+   */
+  set(options: { key: string; value: any; ttl?: Seconds }): Promise<void>;
+
+  /**
+   * Get a cached value or undefined if not set or expired
+   *
+   * @param {string} key
+   * @returns {Promise<any>}
+   * @memberof LocalCacheInterface
+   */
+  get(key: string): Promise<any>;
+
+  /**
+   * Delete a cached value
+   *
+   * @param {string} key
+   * @returns {Promise<void>}
+   * @memberof LocalCacheInterface
+   */
+  delete(key: string): Promise<void>;
+}
+
+interface LocalCacheEntry {
+  value: any;
+  expires?: Date;
+}
+
+export class LocalCache implements LocalCacheInterface {
```

Comment:
general usage question: can there be more than 1 instance of the local cache at the same time? if so, what happens if namespaces are duplicate?

Reply:
Great question and one that's been on my mind as far as usage of the LocalCache. It's technically okay for two instances of LocalCache to share a namespace since they'll both be writing to indexedDB under the hood, but then it's possible to overwrite one another, ie: 
```
localCache1.set({ key: 'foo', value: 'bar'});
localCache2.set({ key: 'foo', value: 'baz'});
```
`localCache2` will overwrite the same `foo` key with `baz` and it probably shouldn't.
---
One of the ways I was thinking about solving this was the following:
1. Instead of instantiating `LocalCache` directly, you have a "LocalCache Factory"
2. To use the LocalCache, you get an instance of the local cache like this: `const localCache1 = await LocalCacheFactory.getLocalCache('namespace')`
3. Use the instance you've been given for your use case (ie user service storage). That `LocalCacheFactory` handles all the namespacing and ensures there are no conflicts between the instances it generates.

Thoughts?

Reply:
I started looking into this approach and I think I'm going to hold off for now since it adds a decent amount of complexity. Currently I was thinking about using a single instance of the cache anyways, but we can implement a factory for it if we find we need it.

Reply:
hi Jason, just saw this. ok sounds good! just talking through the things that came to mind. definitely not a blocker ^_^

`cacheFactory/Registrar` sounds like a good thing to extend when we need it.